PythonRegression/tests/features/steps/stitching_steps.py

```diff
@@ -0,0 +1,130 @@
+from util import static_vals as static
+import threading
+import random
+
+import logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+world.config = {}
+world.responses = {}
+
+TIMEOUT_ERROR = "Thread has not completed, request has timed out"
+
+side_tangle_address = b'SIDE9TANGLE9999999999999999999999999999999999999999999999999999999999999999999999'
+stitching_addresss = b'STITCHING9TRANSACTIONS99999999999999999999999999999999999999999999999999999999999'
+referencing_address = b'REFERENCES9STITCHING9TRANSACTION9999999999999999999999999999999999999999999999999'
+
+@step(r'a stitching transaction is issued on "([^"]*)" with the tag "([^"]*)"')
+def issue_stitching_transaction(step,node,tag):
+    world.config['nodeId'] = node
+    world.config['stitchTag'] = tag
+
+    api = tests.prepare_api_call(node)
+    logger.info('Finding Transactions')
+    side_tangle_transaction = random.choice(static.SIDE_TANGLE_TRANSACTIONS)
+    gtta_transactions = api.get_transactions_to_approve(depth=3)
+
+    trunk = side_tangle_transaction
+    branch = gtta_transactions['branchTransaction']
+    
+    logger.debug('Trunk: ' + str(trunk))
+    logger.debug('Branch: ' + str(branch))
+    
+    bundle = transactions.create_transaction_bundle(stitching_addresss,tag,0)
+    trytes = bundle[0].as_tryte_string()
+    sent_transaction = api.attach_to_tangle(trunk, branch, [trytes], 14)
+    api.broadcast_and_store(sent_transaction.get('trytes'))
+
+    #Finds transaction hash and stores it in world
+    bundlehash = api.find_transactions(bundles=[bundle.hash])
+    world.responses['stitchTransaction'] = bundlehash['hashes'][0]
+
+
+@step(r'check_consistency is called on this transaction')
+def check_stitch_consistency(step):
+    logger.info('Checking consistency of stitching transaction')
+    node = world.config['nodeId']
+    api = tests.prepare_api_call(node)
+
+    transaction = world.responses['stitchTransaction']
+
+    timeout = False
+
+    def consistency(responses):
+        consistency_response = api.check_consistency(tails=[transaction])
+        responses['checkConsistency'] = consistency_response
+        return
+
+    try:
+        r = threading.Thread(target=consistency, args=(world.responses,))
+        r.setDaemon(True)
+        r.start()
+        r.join(timeout=60)
+
+        if r.is_alive():
+            raise ValueError(TIMEOUT_ERROR)
+
+    except ValueError as error:
+        logger.info('Failed')
+        logger.info(error)
+        timeout = True
+
+    if timeout != True:
+        logger.debug('Response: {}'.format(world.responses['checkConsistency']))
+        logger.info('Consistency check completed, request did not time out')
+    else:
+        raise ValueError(TIMEOUT_ERROR)
+
+
+@step(r'the response should return "([^"]*)"')
+def check_response_return_value(step,return_val):
+    logging.info('Checking Return ')
+    response = str(world.responses['checkConsistency']['state'])
```

Comment:
I would put this function in api tests and have `checkConsistency` and `state` configurable via the feature

Reply:
I see you didn't change this.
If you decide to ignore a comment please explain why.
Or is it also in your other PR?

Reply:
This will be replaced in the refactoring PR. I prepared a step already for comparing internal response results so that it can be reused for any api call.